diff --git a/select2.module b/select2.module
index 3312f7c..4653809 100644
--- a/select2.module
+++ b/select2.module
@@ -10,12 +10,12 @@
  */
 function select2_library_info_alter(&$libraries, $extension) {
   if ($extension === 'select2') {
-    $libraries_path = 'libraries/select2';
+    $libraries_path = 'libraries/jquery.select2';
     if (\Drupal::hasService('library.libraries_directory_file_finder')) {
-      $libraries_path = \Drupal::service('library.libraries_directory_file_finder')->find('select2');
+      $libraries_path = \Drupal::service('library.libraries_directory_file_finder')->find('jquery.select2');
     }
     elseif (function_exists('libraries_get_path')) {
-      $libraries_path = libraries_get_path('select2');
+      $libraries_path = libraries_get_path('jquery.select2');
     }
 
     $libraries['select2.min']['js'] = ['/' . $libraries_path . '/dist/js/select2.min.js' => ['minified' => TRUE]];
